### PR TITLE
Expr max minecart speed

### PR DESCRIPTION
**Pull request by DeltaRays**
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprMaxMinecartSpeed.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		if (delta == null) {
+			if (mode == ChangeMode.RESET) getExpr().stream(e).forEach(m -> {
+				if (m instanceof Minecart) ((Minecart) m).setMaxSpeed(0.4);
+			});
+			return;
+		}
+		switch (mode) {
+			case SET:
+				getExpr().stream(e).forEach(m -> {
+					if (m instanceof Minecart) ((Minecart) m).setMaxSpeed(((Number) delta[0]).doubleValue());
```
Comment (ShaneBeee) [inline src/main/java/ch/njol/skript/expressions/ExprMaxMinecartSpeed.java:69]: As stated before on Discord, the condition and void method should not be on the same line.

File: src/main/java/ch/njol/skript/expressions/ExprMaxMinecartSpeed.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		return (mode == ChangeMode.ADD || mode == ChangeMode.REMOVE || mode == ChangeMode.RESET || mode == ChangeMode.SET) ? CollectionUtils.array(Number.class) : null;
+	}
+	
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		if (delta == null) {
+			if (mode == ChangeMode.RESET) getExpr().stream(e).forEach(m -> {
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprMaxMinecartSpeed.java:61]: the IF statement should be on it's own line

File: src/main/java/ch/njol/skript/expressions/ExprMaxMinecartSpeed.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	@Nullable
+	@Override
+	public Number convert(Entity entity) {
+		return entity instanceof Minecart ? ((Minecart) entity).getMaxSpeed() : null;
+	}
+	
+	@Nullable
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		return (mode == ChangeMode.ADD || mode == ChangeMode.REMOVE || mode == ChangeMode.RESET || mode == ChangeMode.SET) ? CollectionUtils.array(Number.class) : null;
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprMaxMinecartSpeed.java:55]: Can you put this statement into a switch?

File: src/main/java/ch/njol/skript/expressions/ExprMaxMinecartSpeed.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+		if (delta == null) {
+			if (mode == ChangeMode.RESET) getExpr().stream(e).forEach(m -> {
+				if (m instanceof Minecart)
+					((Minecart) m).setMaxSpeed(0.4);
+			});
+			return;
+		}
+		switch (mode) {
+			case SET:
+				getExpr().stream(e).forEach(m -> {
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprMaxMinecartSpeed.java:69]: Oh, also, can you switch these to for loops (based on other PRs / feedback to stay consistent)?

File: src/main/java/ch/njol/skript/expressions/ExprMaxMinecartSpeed.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+			case RESET:
+			case SET:
+				return CollectionUtils.array(Number.class);
+			default:
+				return null;
+		}
+	}
+	
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/expressions/ExprMaxMinecartSpeed.java:67]: The switch statement should have a default case that asserts false

**Review by APickledWalrus — CHANGES_REQUESTED** [review]

**Review by ShaneBeee — APPROVED** [review]
Looks good to me

**Review by ShaneBeee — APPROVED** [review]

**Review by APickledWalrus — APPROVED** [review]

**Comment by ShaneBeee** [comment]
Thank you for your contribution

**Pull request merged**